### PR TITLE
Implement a possibility to run TestCafe over https (close #1985)

### LINKED ISSUE #1985: Run testcafe with https protocol
### Are you requesting a feature or reporting a bug?

Requesting a feature (maybe)

### What is the current behavior?

I am unable to start testcafe using https protocol

### What is the expected behavior?

Testcafe should run with ssl certificate (if provided)

### Specify your

* operating system: MacOS 10.13.1
* testcafe version: 0.18.3
* node.js version: 8.7.0

### PR DESCRIPTION
**Pull request by miherlosev**
@AndreyBelym @VasilyStrelyaev

### CODE REVIEW

File: src/cli/argument-parser.js [hunk 1]
```diff
@@ -107,6 +107,7 @@ export default class CLIArgumentParser {
             .option('--ports <port1,port2>', 'specify custom port numbers')
             .option('--hostname <name>', 'specify the hostname')
             .option('--proxy <host>', 'specify the host of the proxy server')
+            .option('--ssl', 'run the TestCafe proxy server over the HTTPS protocol')
```
Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]:
```js
            .option('--ssl', 'run TestCafe proxy server over the HTTPS protocol')
```

Maybe we should rename the flag to `--https` considering the description?

Comment (miherlosev) [reply src/cli/argument-parser.js]: No. Need to change description.
This flag contains ssl options that will passed for https server creation.

Also, this option has same name (`ssl`) for the widely known [http-server](https://www.npmjs.com/package/http-server) module.
Changed description: `specify ssl options for starting TestCafe using https protocol`

Comment (VasilyStrelyaev) [reply src/cli/argument-parser.js]: > `specify ssl options`

So, you can actually specify some options after this flag? Which options?

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/cli/argument-parser.js (hunk 1 above)
Comment (miherlosev) [reply src/cli/argument-parser.js]: Yes, you ca specify a lot of options (all options from [https server options](https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener) and parent classes: [http server iptions](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener), [tls server options](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options)).

See simple example in https://github.com/devexpress/testcafe/issues/1985#issuecomment-389488983

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/cli/argument-parser.js (hunk 1 above)
Comment (VasilyStrelyaev) [reply src/cli/argument-parser.js]: Cool!
As far as I understand, options are not required, so we can write

```js
            .option('--ssl <options>', 'enables access to TestCafe via the HTTPS protocol and allows you to specify SSL options')
```

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/cli/argument-parser.js (hunk 1 above)
Comment (VasilyStrelyaev) [reply src/cli/argument-parser.js]: `specify SSL options for starting TestCafe proxy server over the HTTPS protocol`

Comment (VasilyStrelyaev) [reply src/cli/argument-parser.js]:
```js
            .option('--ssl <options>', 'specify SSL options to run TestCafe proxy server over the HTTPS protocol');
```

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
FPR @VasilyStrelyaev @AndreyBelym

File: src/cli/parse-ssl-options.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        const key          = keyValuePair[0];
+        let value          = keyValuePair[1];
+
+        if (!key || !value)
+            return;
+
+        value = convertToBestFitType(value);
+
+        if (FILE_OPTION_NAMES.includes(key) && value.length < MAX_PATH_LENGTH)
+            value = fs.readFileSync(value);
```
Comment (AndreyBelym) [inline src/cli/parse-ssl-options.js]: On Linux `MAX_PATH_LENGTH` is 4096 (4kB). My SSH private key is about 1.5kB. If I pass my certificate, I will get an error from `fs.readFileSync`. We should check if the file exists, and if it don't, also assume that value is a buffer.

File: src/cli/parse-ssl-options.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import fs from 'fs';
+import os from 'os';
+
+const MAX_PATH_LENGTH = os.type() === 'Windows_NT' ? 260 : Infinity;
```
Comment (AndreyBelym) [inline src/cli/parse-ssl-options.js]: `MAX_PATH_LENGTH  === Infinity` means you can't pass a certificate content for properties that accept both file paths and file contents.

**Comment by miherlosev** [comment]
ping @VasilyStrelyaev

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**